rust/operator-binary/src/druid_controller.rs

```diff
@@ -366,6 +369,13 @@ fn build_rolegroup_config_map(
                     if let Some(endpoint) = conn.endpoint() {
                         transformed_config.insert(S3_ENDPOINT_URL.to_string(), Some(endpoint));
                     } // TODO make code nicer
+
+                    let path_style_access =
+                        conn.access_style.clone().unwrap_or_default() == S3AccessStyle::Path;
+                    transformed_config.insert(
+                        S3_PATH_STYLE_ACCESS.to_string(),
+                        Some(path_style_access.to_string()),
+                    );
```

Comment:
I would prefer to use a `match` expression for cases like this, where we want to flatten an enum into a (series of) booleans.

That way, we get a compilation error if a future change adds more options that need to be handled.

Reply:
I had the same though process. But i came to the conclusion that Druid doesn't support something else so adding a third accessStyle would require extensive implementation and testing anyway. But i agree with you and will convert it back to a match

Reply:
Yep, I'm not saying that things will just work when new options are added, just that it helps to highlight the problem.